lib/modules/versioning/azure-rest-api/index.ts

```diff
@@ -0,0 +1,49 @@
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'azure-rest-api';
+export const displayName = 'azure-rest-api';
+
+export const urls = [
+  'https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-versioning',
+];
+
+export const supportsRanges = false;
+
+const AZURE_REST_API_VERSION_REGEX = regEx(
+  /^(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})(?<prerelease>-[a-z]+)?$/
+);
+
+class AzureRestApiVersioningApi extends GenericVersioningApi {
+  protected _parse(version: string): GenericVersion | null {
+    if (!version) {
+      return null;
+    }
+
+    const matchGroups = AZURE_REST_API_VERSION_REGEX.exec(version)?.groups;
+
+    if (!matchGroups) {
+      return null;
+    }
+
+    const { year, month, day, prerelease } = matchGroups;
+
+    return {
+      release: [parseInt(`${year}${month}${day}`), 0, 0],
```

Comment:
I think combining these into the major version is the right move, because it may or may not be a breaking change. It does lead to odd version numbers in commit messages: https://github.com/skyfrk-bot/test-renovate/pull/12.

I wonder if it's possible to split these versions using a custom `commitMessageExtra`?